app/assets/javascripts/darkswarm/directives/validate_stock_quantity.js.coffee

```diff
@@ -0,0 +1,17 @@
+Darkswarm.directive "validateStockQuantity", ->
+  restrict: 'A'
+  require: "ngModel"
+
+  link: (scope, element, attr, ngModel) ->
+    ngModel.$parsers.push (selectedQuantity) ->
+      if parseInt(selectedQuantity) > scope.available_quantity()
+        ngModel.$setValidity('stock', false);
+      else
+        ngModel.$setValidity('stock', true);
```

Comment:
```javascript
      valid? = parseInt(selectedQuantity) > scope.available_quantity()
      ngModel.$setValidity('stock', valid?);
```

Reply:
re parseInt - what happens if user types a letter in the quantity field?

Reply:
I've made this more explicit :+1: